internal/mutators/podtemplatespec.go

```diff
@@ -0,0 +1,113 @@
+package mutators
+
+import (
+	"fmt"
+
+	"github.com/kanopy-platform/hedgetrimmer/internal/limitrange"
+	"github.com/kanopy-platform/hedgetrimmer/internal/quantity"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type PodTemplateSpec struct{}
+
+func NewPtsMutator() PodTemplateSpec {
+	return PodTemplateSpec{}
+}
+
+func (p PodTemplateSpec) Mutate(inputPts corev1.PodTemplateSpec, lri corev1.LimitRangeItem) (corev1.PodTemplateSpec, error) {
+	pts := *inputPts.DeepCopy()
+
+	if !limitrange.IsLimitRangeTypeContainer(lri) {
+		return pts, fmt.Errorf("expected LimitRangeItem type Container, got %q instead", lri.Type)
+	}
+
+	limitRangeMemory := limitrange.NewConfig(lri, corev1.ResourceMemory)
+
+	if err := setAndValidateResourceRequirements(pts.Spec.InitContainers, limitRangeMemory); err != nil {
+		return pts, err
+	}
+
+	if err := setAndValidateResourceRequirements(pts.Spec.Containers, limitRangeMemory); err != nil {
+		return pts, err
+	}
+
+	return pts, nil
+}
+
+func setAndValidateResourceRequirements(containers []corev1.Container, limitRangeMemory limitrange.Config) error {
+	for idx := range containers {
+		container := &containers[idx]
+		if err := setMemoryRequest(container, limitRangeMemory); err != nil {
+			return err
+		}
+
+		if err := setMemoryLimit(container, limitRangeMemory); err != nil {
+			return err
+		}
+
+		if err := validateMemoryRequirements(*container, limitRangeMemory); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateMemoryRequirements(container corev1.Container, limitRangeMemory limitrange.Config) error {
+	memoryRequest := container.Resources.Requests.Memory()
+	memoryLimit := container.Resources.Limits.Memory()
+
+	if memoryRequest.IsZero() || memoryLimit.IsZero() {
+		return fmt.Errorf("memory request (%s) and limit (%s) must be set", memoryRequest.String(), memoryLimit.String())
+	}
+
+	if limitRangeMemory.HasMaxLimitRequestRatio {
+		ratio := quantity.Div(*memoryLimit, *memoryRequest)
+		if ratio.Cmp(limitRangeMemory.MaxLimitRequestRatio) == 1 {
```

Comment:
Is this the same precision as the [limitranger admission controller](https://github.com/kubernetes/kubernetes/blob/462f41212a4e5ded8974784038f681a08dded1be/plugin/pkg/admission/limitranger/admission.go#L354)? 

We want to be sure we don't let in a request that the LimitRanger will then reject.

Reply:
I'll take a look at .Value() and ensure our scale matches limitranger's